subscription-service/src/test/kotlin/com/egm/stellio/subscription/service/NotificationServiceTests.kt

```diff
@@ -404,4 +422,120 @@ class NotificationServiceTests {
 
         verify(1, postRequestedFor(urlPathEqualTo("/notification")))
     }
+
+    @Test
+    fun `it should notify the subscriber and return entities without system attributes if sysAttrs is false`() = runTest {
+
+        val payload = mapOf(
+            "id" to "urn:ngsi-ld:Subscription:1",
+            "type" to JsonLdUtils.NGSILD_SUBSCRIPTION_TERM,
+            "entities" to listOf(
+                mapOf(
+                    "id" to apiaryId,
+                    "type" to "Apiary",
+                    "@context" to APIC_COMPOUND_CONTEXT
+                )
+            ) ,
+            "notification" to mapOf(
+                "endpoint" to mapOf(
+                "accept" to "application/ld+json",
+                "uri" to "http://localhost:8089/notification"),
+                "sysAttrs" to false
+            )
+        )
+
+
+        val subscription = ParsingUtils.parseSubscription(payload, emptyList()).shouldSucceedAndResult()
+        val expandedEntity = expandJsonLdEntity(entity)
+
+        coEvery {
+            subscriptionService.getMatchingSubscriptions(any(), any(), any())
+        } returns listOf(subscription).right()
+        coEvery { subscriptionService.updateSubscriptionNotification(any(), any(), any()) } returns 1
+
+        stubFor(
+            post(urlMatching("/notification"))
+                .willReturn(ok())
+        )
+
+        notificationService.notifyMatchingSubscribers(
+            expandedEntity,
+            setOf(NGSILD_NAME_PROPERTY),
+            ATTRIBUTE_UPDATED
+        ).shouldSucceedWith {
+            val entity = it[0].second.data[0]
+            assertFalse(entity.containsKey("createdAt"))
+            assertFalse(entity.containsKey("modifiedAt"))
+
+            for ((key, value) in entity) {
+                if (value !is Map<*, *>) {
+                    continue
+                }
+                 assertFalse((value as Map<*, *>).containsKey("createdAt"))
+                    assertFalse((value as Map<*, *>).containsKey("modifiedAt"))
+
+            }
+        }
+    }
+
+    @Test
+    fun `it should notify the subscriber and return entities with system attributes if sysAttrs is true`() = runTest {
+
+        val payload = mapOf(
+            "id" to "urn:ngsi-ld:Subscription:1",
+            "type" to JsonLdUtils.NGSILD_SUBSCRIPTION_TERM,
+            "entities" to listOf(
+                mapOf(
+                    "id" to apiaryId,
+                    "type" to "Apiary",
+                    "createdAt" to  "2024-02-13T18:15:00Z",
+                    "modifiedAt" to "2024-02-13T18:16:00Z",
+                    "name" to mapOf(
+                        "type" to "Property",
+                        "value" to "ApiarySophia",
+                        "createdAt" to "2024-02-13T18:15:00Z",
+                        "modifiedAt" to "2024-02-13T18:16:00Z"
+                    ),
+                    "@context" to APIC_COMPOUND_CONTEXT
```

Comment:
you should not have all these things in a subscription. in the subscription, you only say what you want to be notified about, you don't put an entity in the subscription (see 5.2.12).